### DISCUSSION TITLE
Best Practices for bundle tests -- simple-phpunit or simply phpunit?

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#58954] [Q&A]
Is it still the best practice to use symfony/phpunit-bridge and vendor/bin/simple-phpunit?  Or with PHPUnit 11 is that no longer necessary?  

Or does it have to do with making it easier to get Symfony services?

The docs aren't clear:

> The test suite must be executable with a simple phpunit command run from a sample application;

https://symfony.com/doc/current/bundles/best_practices.html#tests

I've been trying to find a relatively simple bundle with the latest PHPUnit tests, but a bundle with a Controller and a Command and maybe a TwigExtension.  Any suggestions?

**Comment by stof** [comment]
If your package targets PHP 8.2+, my suggestion is to use PHPUnit 11, with its builtin deprecation reporting. And if you rely on the ClockMock or DnsMock of the phpunit-bridge, use version 7.2 of the bridge where those are compatible with PHPUnit 11.
We should probably provide a recipe for PHPUnit 11+ with the appropriate config to report deprecations similar to what we do with the bridge, as this requires a non-default config.

If your package supports older PHP versions, use PHPUnit 9.6 and `symfony/phpunit-bridge`.
Personally, I always put PHPUnit as a dev requirement and I don't use the `simple-phpunit` executable, as phpunit has to be compatible with our dependencies anyway as it runs in the same PHP process. This only requires an explicit registration of the SymfonyTestListener in the phpunit.xml config file while `simple-phpunit` does it automatically.

**Comment by tacman** [comment]
A recipe would be great.

My bundles are all php 8.2+ and symfony 6.4+, so phpunit 11 should work, I
just need an example to follow.
